lib/api/completeMultipartUpload.js

```diff
@@ -182,7 +182,13 @@ function completeMultipartUpload(authInfo, request, log, callback) {
                 // in numerical order
                 // return an error
                 if (Number.parseInt(part.PartNumber[0], 10) !== i + 1) {
-                    return next(errors.InvalidPartOrder);
+                    // If not in order return InvalidPartOrder
+                    if (jsonList.Part.some(partVal =>
```

Comment:
Instead of looping over the list within the loop, can't we check that the part is just not greater than the max part and return InvalidPart if it is?